src/Text/Pandoc/Readers/Org/Blocks.hs

```diff
@@ -64,11 +68,22 @@ import qualified Text.Pandoc.Walk as Walk
 -- | Get a list of blocks.
 blockList :: PandocMonad m => OrgParser m [Block]
 blockList = do
-  headlines      <- documentTree blocks inline
+  fHeadlineTree  <- documentTree blocks inline
   st             <- getState
-  headlineBlocks <- headlineToBlocks $ runF headlines st
-  -- ignore first headline, it's the document's title
-  return . drop 1 . B.toList $ headlineBlocks
+  let headlineTree = runF fHeadlineTree st
```

Comment:
I haven't thought this through, but would it be possible to move this into `headlineToBlocks`? It would be nice if we could keep all tree-walking/modifications in that file.